solutions/usaco-663.mdx

```diff
@@ -24,20 +35,59 @@ int main()
 	freopen("square.out", "w", stdout);
 	int x1, y1, x2, y2;
 	int x3, y3, x4, y4;
-	int left, right, top, bottom;
 	cin >> x1 >> y1 >> x2 >> y2 >> x3 >> y3 >> x4 >> y4;
+	int left, right, top, bottom;
+	// find the sides of the smallest rectangle covering both pastures
 	left = min(x1, x3);
 	right = max(x2, x4);
 	bottom = min(y1, y3);
 	top = max(y2, y4);
+	/*
+	 * the smallest square will need a side length
+	 * that is the maximum of the side lengths of the rectangle
+	 */
 	int side = max(right - left, top - bottom);
 	cout << side * side << endl;
 	return 0;
 }
+
 ```
 
 </CPPSection>
 
+<JavaSection>
+
+```java
+import java.io.*;
+import java.util.*;
+
+public class SquarePasture
```

Comment:
```suggestion
public class Square
```

Reply:
is there a reason for this? idk java soooo....

Reply:
so it matches the problem name

Reply:
Wait yeah I feel like SquarePasture would be better

Reply:
Oh nvm that's what Sans suggested. I think SquarePasture is a bit more descriptive of the problem.

Reply:
sans suggested changing SquarePasture -> Square

Reply:
if you're looking at the solution probably have an idea of what the problem is so

Reply:
> so it matches the problem name

the problem name is Square Pasture though...

Reply:
Oh ok I'm dumb yeah I think yours is better then

Reply:
i already changed it an i don't think it really matters, so i'll just leave it